### PR TITLE
Add Tristate method to TextDecoration.State

### PR DESCRIPTION
**Pull request by KingOfSquares**
Felt like a naturally missing piece

### CODE REVIEW

File: api/src/main/java/net/kyori/adventure/text/format/TextDecoration.java [hunk 1]
```diff
@@ -166,5 +167,16 @@ public String toString() {
...
+
+    /**
+     * Gets a state from a {@link net.kyori.adventure.util.TriState}.
+     *
+     * @param flag the tristate
+     * @return the state
+     * @since 4.10.0
+     */
+    public static @NotNull State byTriState(final @NotNull TriState flag) {
+      return byBoolean(flag.toBoolean());
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/text/format/TextDecoration.java:179]: Should have a require not null check on the flag

**Review by kezz — CHANGES_REQUESTED** [review]

File: api/src/main/java/net/kyori/adventure/text/format/TextDecoration.java [hunk 2]
```diff
@@ -166,5 +169,17 @@ public String toString() {
...
+    /**
+     * Gets a state from a {@link net.kyori.adventure.util.TriState}.
+     *
+     * @param flag the tristate
+     * @return the state
+     * @since 4.10.0
+     */
+    public static @NotNull State byTriState(final @NotNull TriState flag) {
+      requireNonNull(flag);
+      return byBoolean(flag.toBoolean());
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/format/TextDecoration.java:182]: Think I'd prefer a `switch` here.

**Review by kashike — CHANGES_REQUESTED** [review]

**Review by kezz — APPROVED** [review]

**Comment by BomBardyGamer** [comment]
Wouldn't it also be useful to add a `asTriState` conversion method, so you can convert a state to its tri state representation?

**Comment by kezz** [comment]
> Wouldn't it also be useful to add a `asTriState` conversion method, so you can convert a state to its tri state representation?

I don't think so. This comes back to the reason they aren't the same in the first place. A `TriState` will never have its values change but a `TextDecoration.State` might.

**Review by kashike — APPROVED** [review]

**Review by zml2008 — APPROVED** [review]

**Pull request merged**